### PR TITLE
Jsonb TCK test removal

### PR DESCRIPTION
**Pull request by Verdent**
**Fixes Issue**
TCK test duplication. All of these removed tests are already present in the standalone version in JSON-B repository.

**Describe the change**
All the JSON-B TCK tests are removed. Only CDI tests and newly created pluggability tests remain.

CC @alwin-joseph @gurunrao @scottmarlow

### CODE REVIEW

File: install/jsonb/bin/ts.jte [hunk 1]
```diff
@@ -50,8 +50,8 @@ jsonb.classes=
 #
 #                                Added to ts.run.classpath below.
 ###############################################################################
-#jsonb.resources=${ts.home}/src/com/sun/ts/tests/jsonb/resources
```
Comment (gurunrao) [inline install/jsonb/bin/ts.jte:53]: the changes to install/jsonb/bin/ts.jte and install/jsonb/bin/build.xml are not required, these files will be removed as part of standalone tck remove PR(https://github.com/eclipse-ee4j/jakartaee-tck/pull/848)

**Comment by scottmarlow** [comment]
[We discussed a possible problem today with removing jsonb tests that validate Platform requirement.](https://www.eclipse.org/lists/jsonb-dev/threads.html#00086)

If we [look at an EE 9.1 Platform TCK test run](https://ci.eclipse.org/jakartaee-tck/job/9.1/job/eftl-jakartaeetck-run-910/lastCompletedBuild/testReport/com.sun.ts.tests.jsonb.api.annotation/AnnotationTest/), we will see Platform TCK test results like:
```
testJsonbArrayTypeSerializer_from_appclient
testJsonbArrayTypeSerializer_from_ejb
testJsonbArrayTypeSerializer_from_jsp
testJsonbArrayTypeSerializer_from_servlet
``` 

The [requirement for running the appclient, ejb, jsp, servlet vehicles/containers are expressed here](https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/install/jakartaee/other/vehicle.properties#L238).

I think the action that we need to take is to ensure that [we satisfy the Platform requirements for JSONB which (from EE 9.1)](https://jakarta.ee/specifications/platform/9.1/jakarta-platform-spec-9.1.html#a2713) are:

> In a full Jakarta EE product, all Jakarta EE application client containers, web containers, and enterprise beans containers are required to support the JSON-B API.

FYI, some of the new Standalone TCKs are validating Platform requirements also, we just didn't do that with the JSONB TCK.

In summary, our actions to take are:
- Ensure that the JSON-B API is tested from an application client container.
- Ensure that the JSON-B API is tested from an Enterprise Beans container.
- Ensure that the JSON-B API is tested from a Server Pages container.
- Ensure that the JSON-B API is tested from a Servlet container.

Our options for ensuring ^ may be taking by having a set of tests that could be all of the Platform TCK JSON-B tests that are currently run on the mentioned containers, or a smaller subset of tests than that.  I suggest that we get input from anyone willing to give it, on which such tests to keep.  

I'll give others a chance to respond before I give me feedback, other than to say my experience with reverse engineering the TCK test AssertionIDs to the actual Specification requirements, is that IMO the requirement gathering process can be slow.  

Some background materials on [AssertionIDS is here.](https://github.com/eclipse-ee4j/jakartaee-tck/wiki/Requirements-for-adding-Specification-assertions-to-TCK-tests)

In case it helps the conversation, here are some options to consider:
1. Keep all of the JSON-B tests in the Platform TCK to run on the required containers mentioned above.
2. Keep a minimal number of JSON-B tests in the Platform TCK to run on the required containers mentioned above.  Also require the Standalone JSON-B TCK to be run against the JSON-B impl used by Platform implementation.
3. Keep some but not all of JSON-B tests in the Platform TCK to run on the required containers mentioned above.  Also require the Standalone JSON-B TCK to be run against the JSON-B impl used by Platform implementation.
4. Explain another option not included here.

**Comment by gurunrao** [comment]
@scottmarlow - By retaining 2-4 test from https://github.com/eclipse-ee4j/jakartaee-tck/tree/master/src/com/sun/ts/tests/jsonb/api, will that help address https://github.com/eclipse-ee4j/jakartaee-tck/pull/849#issuecomment-1049376598 comments?

**Comment by m0mus** [comment]
@scottmarlow We've discussed options already and decided to choose option 2 from your list. I consider this question closed. I don't want us to go in circles.

**Comment by scottmarlow** [comment]
> @scottmarlow We've discussed options already and decided to choose option 2 from your list. I consider this question closed. I don't want us to go in circles.

@m0mus while we did discuss eliminating the duplicate tests as a goal, in that discussion we also stated that ensuring that Platform requirements are still validated is important.  

IMO, option 2 is fine for us to choose for JSON-B.

**Comment by scottmarlow** [comment]
> @scottmarlow - By retaining 2-4 test from https://github.com/eclipse-ee4j/jakartaee-tck/tree/master/src/com/sun/ts/tests/jsonb/api, will help address [#849 (comment)](https://github.com/eclipse-ee4j/jakartaee-tck/pull/849#issuecomment-1049376598) comments.

:+1: 

Should we keep the (duplicate) signature tests for JSON-B in the Platform TCK as extra validation of the SPEC API?  

We are going to require EE implementations to also pass the JSON-B Standalone TCK, which contains the signature tests, so IMO it it would be okay to remove the Platform TCK copy of JSON-B signature map testing.

**Comment by lukasj** [comment]
> Should we keep the (duplicate) signature tests for JSON-B in the Platform TCK as extra validation of the SPEC API?

Yes - to make sure that what the platform provides at the runtime of each container (appclient/webapp/ejb) actually has the right API. Should the TCK rely solely on the results form the standalone test/certification, there is no guarantee that ie webcontainer's classloader does no magic to these APIs for webapps (just my point of view...)

**Comment by scottmarlow** [comment]
> > Should we keep the (duplicate) signature tests for JSON-B in the Platform TCK as extra validation of the SPEC API?
> 
> Yes - to make sure that what the platform provides at the runtime of each container (appclient/webapp/ejb) actually has the right API. Should the TCK rely solely on the results form the standalone test/certification, there is no guarantee that ie webcontainer's classloader does no magic to these APIs for webapps (just my point of view...)

:+1:

**Comment by Verdent** [comment]
Signature tests should be still present and were not removed by this PR, so I think there is no need to do any changes in this PR in this matter.

**Comment by Verdent** [comment]
Do you want me to do any changes to this PR?

**Comment by gurunrao** [comment]
> > Do you want me to do any changes to this PR?
> 
> No further changes are needed beyond what reviewers might request when reviewing this pull request soon.
> 
> I did some testing with this pr on [Platform TCK Jenkins](https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck-scottmarlow/job/jsonb-tck-removal/2).
> 
> I verified that the below tests were run which does ensure that the JSON-B API can be used in the (Platform SPEC) required containers:
> 
>     * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest1_from_appclient
> 
>     * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest1_from_ejb
> 
>     * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest1_from_jsp
> 
>     * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest1_from_servlet
> 
>     * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest2_from_appclient
> 
>     * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest2_from_ejb
> 
>     * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest2_from_jsp
> 
>     * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest2_from_servlet
> 
>     * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest3_from_appclient
> 
>     * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest3_from_ejb
> 
>     * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest3_from_jsp
> 
>     * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest3_from_servlet
> 
> 
> And the following Platform TCK tests failed:
> 
>     * jsonb.cdi.customizedmapping.adapters.AdaptersCustomizationCDITest.testCDISupport_from_ejb
> 
>     * jsonb.cdi.customizedmapping.adapters.AdaptersCustomizationCDITest.testCDISupport_from_jsp
> 
>     * jsonb.cdi.customizedmapping.adapters.AdaptersCustomizationCDITest.testCDISupport_from_servlet
> 
>     * jsonb.cdi.customizedmapping.serializers.SerializersCustomizationCDITest.testCDISupport_from_ejb
> 
>     * jsonb.cdi.customizedmapping.serializers.SerializersCustomizationCDITest.testCDISupport_from_jsp
> 
>     * jsonb.cdi.customizedmapping.serializers.SerializersCustomizationCDITest.testCDISupport_from_servlet

CDI tests are dependent on latest Yasson integration with glassfish 7.Latest Yasson RC has dependency with CDI fixed.
We will re-test, post dependencies are sorted at GF.

**Comment by scottmarlow** [comment]
> > > Do you want me to do any changes to this PR?
> > 
> > 
> > No further changes are needed beyond what reviewers might request when reviewing this pull request soon.
> > I did some testing with this pr on [Platform TCK Jenkins](https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck-scottmarlow/job/jsonb-tck-removal/2).
> > I verified that the below tests were run which does ensure that the JSON-B API can be used in the (Platform SPEC) required containers:
> > ```
> > * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest1_from_appclient
> > 
> > * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest1_from_ejb
> > 
> > * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest1_from_jsp
> > 
> > * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest1_from_servlet
> > 
> > * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest2_from_appclient
> > 
> > * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest2_from_ejb
> > 
> > * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest2_from_jsp
> > 
> > * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest2_from_servlet
> > 
> > * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest3_from_appclient
> > 
> > * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest3_from_ejb
> > 
> > * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest3_from_jsp
> > 
> > * jsonb.pluggability.jsonbprovidertests.jsonbProviderTest3_from_servlet
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And the following Platform TCK tests failed:
> > ```
> > * jsonb.cdi.customizedmapping.adapters.AdaptersCustomizationCDITest.testCDISupport_from_ejb
> > 
> > * jsonb.cdi.customizedmapping.adapters.AdaptersCustomizationCDITest.testCDISupport_from_jsp
> > 
> > * jsonb.cdi.customizedmapping.adapters.AdaptersCustomizationCDITest.testCDISupport_from_servlet
> > 
> > * jsonb.cdi.customizedmapping.serializers.SerializersCustomizationCDITest.testCDISupport_from_ejb
> > 
> > * jsonb.cdi.customizedmapping.serializers.SerializersCustomizationCDITest.testCDISupport_from_jsp
> > 
> > * jsonb.cdi.customizedmapping.serializers.SerializersCustomizationCDITest.testCDISupport_from_servlet
> > ```
> 
> CDI tests are dependent on latest Yasson integration with glassfish 7.Latest Yasson RC has dependency with CDI fixed. We will re-test, post dependencies are sorted at GF.

:+1:

**Review by scottmarlow — APPROVED** [review]

**Comment by gurunrao** [comment]
if there are no more changes required, shall we merge the PR?

**Pull request merged**